### PR TITLE
Adds proper item stack support to heretic transmutation + gives some generally better feedback for certain failed transmutations

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Adds proper stack support to heretic items instead of hard-coding it in two spots. 
- Adds some generally better feedback for some failed heretic rituals
- Fixes some weird bugs with heretic rituals vanishing stuff. 
- Adds some better documentation to the ritual and rune stuff, and moves some of the ritual vars up a level. 

## Why It's Good For The Game

I wanted to clean up the rituals a bit post-merge, so yeah, here. 
Also fixes some bugs in the process and improves documentation. 

## Changelog

:cl: Melbert
fix: Fixes a few potential issues with heretic rituals.
code: Adds proper stack support to heretic rituals. Also allows for more specific "ritual failure" feedback. 
/:cl:

### CODE REVIEW

File: code/modules/antagonists/heretic/knowledge/general_side.dm [hunk 1]
```diff
@@ -36,6 +36,7 @@
 		CRASH("Heretic datum didn't have a hunt_and_sacrifice knowledge learned, what?")
 
 	if(!target_finder.obtain_targets(user))
+		loc.balloon_alert("ritual failed, no targets!")
```
Comment (Mothblocks) [inline code/modules/antagonists/heretic/knowledge/general_side.dm:39]:
```suggestion
		loc.balloon_alert(user, "ritual failed, no targets!")
```

Runtime 😛

Comment (MrMelbert) [reply code/modules/antagonists/heretic/knowledge/general_side.dm:39]: don't code at 3am >.>;;

File: code/modules/antagonists/heretic/heretic_knowledge.dm [hunk 2]
```diff
@@ -124,6 +124,17 @@
 		if(isliving(sacrificed))
 			continue
 
+		if(istype(sacrificed, /obj/item/stack))
```
Comment (Ryll-Ryll) [inline code/modules/antagonists/heretic/heretic_knowledge.dm:127]:
```suggestion
		if(isstack(sacrificed))
```

File: code/modules/antagonists/heretic/transmutation_rune.dm [hunk 3]
```diff
@@ -103,11 +104,17 @@
...
 
-				// This item is a valid type.
-				// Add it to our selected atoms list
-				// and decrement the value of our requirements list
+				// This item is a valid type. Add it to our selected atoms list.
 				selected_atoms |= nearby_atom
-				requirements_list[req_type]--
+				// If it's a stack, we gotta see if it has more than one inside,
+				// as our requirements may want more than one item of a stack
+				if(istype(nearby_atom, /obj/item/stack))
```
Comment (Ryll-Ryll) [inline code/modules/antagonists/heretic/transmutation_rune.dm:111]:
```suggestion
				if(isstack(nearby_atom))
```

**Review by Ryll-Ryll — APPROVED** [review]

**Pull request merged**